### ISSUE TITLE
Model detects COVID-19 or Data Source?

### ISSUE THREAD

**Issue by ajaymaity** [lindawangg/COVID-Net#20]
Given the datasource is different for the two classes: COVID and Pneumonia/Normal, how do you validate that the model doesn't classify the data source, but actually classifies the presence of COVID-19?

**Comment by ogencoglu** [comment]
+1

This is the most important issue.

**Comment by ashcrok** [comment]
It is something that I already experiment on with a model of my own, using similar data, and it seems that the model discriminates between the classes by their sources only. This is a very important question to be pursued.

**Comment by lindawangg** [comment]
Agreed. We use explainability to better understand the model and why it made its decision. We are also speaking with different hospitals as it is also important to get input from hospitals and to test these models in clinical settings for further verification.

**Comment by OGordon100** [comment]
Also, what stage of COVID are these photos taken at? 

Is it effectively pneumonia vs severe pneumonia? Is it only those hospitalised? Is it early symptoms?

Is the author implying this only has use as an accurate testing tool when the patient has developed severe symptoms requiring severe medical intervention? (and if so, what's the actual use case?)

Indeed, there appears to be little-to-no preprocessing of the dataset going on - there are also likely to be differences in how the scans were taken depending on data class, which would then incorrectly lead to correct classification.

Further, the dataset is massively weighted, so looking at accuracy/TP/FP in the paper is misleading

I heavily doubt the correctness of this study. It just seems like a "hey, let's give some random data to a CNN and try publish it" type deal. Indeed, in the current environment this approach is actually quite dangerous. We as the scientific community really need to "know our place" at the moment - we should not be applying our knowledge to medicine without an _extremely_ good cause - the general public do not appreciate the difference between published research and the arxiv - they just see "hey science said this". This has led to incredible issues with medarxiv and national journals running articles on shoddy science, which is immediately rebunked by the relevant, knowledgeable community.

**Comment by lindawangg** [comment]
We appreciate your feedback and want to make some clarifications with related to it.

We made it very clear and transparent in the paper where the datasets were collected, and make it also very clear in the GitHub Repo.  Specifically, all non-COVID samples are from the dataset collected by the Radiological Society of North America (https://www.kaggle.com/c/rsna-pneumonia-detection-challenge) and all COVID-19 samples came from the database collected by Cohen et al (https://github.com/ieee8023/covid-chestxray-dataset). Cohen et al. also included a paper, https://arxiv.org/pdf/2003.11597.pdf, which states the goal of their dataset collection and additional notes about each COVID-19 sample in their dataset, including the state of the patient at the time of the xray. Although not perfectly balanced, the dataset has covid-19 samples collected at different stages.

As we stated both in the paper and many times in the readme, this is not a production ready system. In order for this model to be accurately evaluated, clinical studies NEED to be conducted. In addition, if models like these are to be deployed, they are not meant to replace radiologists. Rather, they are meant to aid them in their decisions and alleviate their workload, which is very important given the current stress on the healthcare system as a result of the COVID-19 pandemic. Exactly how that would be done will involve many iteration cycles with feedback from the doctors and healthcare workers, which we are currently doing with their generous support of the initiative.

In this updated paper, https://github.com/lindawangg/COVID-Net/blob/master/assets/COVID_Netv2.pdf, under Implementation details, we have mentioned that we used random combinations of augmentation techniques, including intensity shift and rotation, to account for discrepancies between xrays from different sources. Of course, this is not perfect and still a research effort. We have also stated in the paper that we use explainability to further investigate and explore how COVID-Net makes predictions. By analyzing critical factors of how the model is making decisions, we can understand what the model has learned. In Figure 6 of the paper, the critical factors identified were on the chest area and not looking at the letters for instance, which is important to verify that the model is learning from the current cues and visual indicators.

Both at the beginning and end of the paper, as well as in the repository where press has been linking to, we have stated that this is not production ready and will involve further research effort both by our team and other researchers, as well as doctors and healthcare workers. As the scientific community, we do indeed "know our place", and that "place" is to use our expertise as well to work with those who have clinical expertise such as medical professionals and clinical scientists, since they do not possess the machine learning expertise to turn this into a reality to help the global community at this critical time unless we all work together on the problem.  To not take initiative at a time like this and to watch and hope others will deal with the problem to their own accord without speaking to them and working with them on a solution for the problem is not the right thing to do.  To narrow our perspective on problems only when others have brought it up rather than take initiative to explore parallel solutions is also not the right thing to do.

The goal of the project is to raise awareness and if hospitals are interested, they can contact us so we can work together.  Many have indeed reached out to provide their expertise and support for this initiative and we are proud to continue to work with them.
